### PR TITLE
AddingDocumentsPage

### PR DESCRIPTION
**Pull request by AlexKaschuk**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/api/documentsApi.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+}
+
+const getById = async (id: number) => {
+    const response: DocumentPost = await (await Api.get(`MethodicDocuments/${id}`)).data;
+    return response;
+};
+
+const getAll = async () => {
+    const { data } = await Api.get("MethodicDocuments");
+
```
Comment (vovikboyko) [inline src/api/documentsApi.ts:69]: empty row

File: src/api/documentsApi.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    const file = dataURLtoFile(response, fileName);
+    const anchor = window.document.createElement('a');
+    anchor.href = response;
+    anchor.download = fileName;
+    document.body.appendChild(anchor);
+    anchor.click();
+    document.body.removeChild(anchor);
+    window.URL.revokeObjectURL(anchor.href);
+    return response;
+}
```
Comment (vovikboyko) [inline src/api/documentsApi.ts:122]: need empty row

File: src/api/documentsApi.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+const put = async (id: number, data: DocumentPost) => {
+    const response = await Api.put(`MethodicDocuments/${id}`, data).then(response => {
+        notificationLogic('success', successfulEditAction("Документ"));
+    })
+        .catch(error => {
+            notificationLogic('error', error.response.data.value);
+        });
+
```
Comment (vovikboyko) [inline src/api/documentsApi.ts:130]: empty row

File: src/api/documentsApi.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+            notificationLogic('error', error.response.data.value);
+        });
+    return response;
+};
+
+export const TypePostParser = (Type: MethodicDocumentType): number => {
+    if (Type.value === "legislation") return 0;
+    if (Type.value === "Methodics") return 1;
+    return 2;
+};
```
Comment (vovikboyko) [inline src/api/documentsApi.ts:148]: need empty row

File: src/api/documentsApi.ts [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+export const TypeGetParser = (Type: number): string => {
+    if (Type === 0) return "Нормативний акт";
+    if (Type === 1) return "Методичний документ";
+    if (Type === 2) return "Інше";
+    return "Не визначено";
+};
+
```
Comment (vovikboyko) [inline src/api/documentsApi.ts:155]: empty row

**Review by MykhailoKlapchuk — APPROVED** [review]

**Review by vovikboyko — APPROVED** [review]

File: src/api/documentsApi.ts [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+export type Document = {
+    id: number;
+    name: string;
+    type: string;
+    organization: string;
+    description: string;
+    date: string;
+    fileName: string | null;
+}
```
Comment (LanchevychMaxym) [inline src/api/documentsApi.ts:13]: It's better to extract this type to separate Model file

File: src/api/documentsApi.ts [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+export type DocumentPost = {
+    id: number;
+    name: string;
+    type: number;
+    organization: Organization;
+    description: string;
+    date: string;
+    fileName: string | null;
+}
```
Comment (LanchevychMaxym) [inline src/api/documentsApi.ts:46]: The same issue.

File: src/api/documentsApi.ts [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+        const ascii = binaryString.charCodeAt(i);
+        bytes[i] = ascii;
+    };
+    const blob = new Blob([bytes], { type: "application/pdf" });
+    const link = window.URL.createObjectURL(blob);
+    return link;
+};
+
+const post = async (data: any) => {
+    console.log(data);
```
Comment (LanchevychMaxym) [inline src/api/documentsApi.ts:93]: It's unnecessary

**Review by LanchevychMaxym — CHANGES_REQUESTED** [review]

File: src/api/documentsApi.ts (hunk 8 above)
Comment (LanchevychMaxym) [reply src/api/documentsApi.ts:93]: I mean "console.log"

**Review by tetyanakutsko — APPROVED** [review]

File: src/api/documentsApi.ts (hunk 8 above)
Comment (AlexKaschuk) [reply src/api/documentsApi.ts]: Done

File: src/api/documentsApi.ts (hunk 7 above)
Comment (AlexKaschuk) [reply src/api/documentsApi.ts]: Done

File: src/api/documentsApi.ts (hunk 6 above)
Comment (AlexKaschuk) [reply src/api/documentsApi.ts]: Done

**Review by LanchevychMaxym — APPROVED** [review]

**Pull request merged**